src/globals.h

```diff
@@ -101,6 +101,44 @@ extern int vb6Round(double x);
  */
 extern double vb6Round(double x, int decimals);
 
+
+// utilities for stringindex_t
+extern std::vector<std::string> g_LevelString;
+extern size_t g_numWorldString;
+
+inline std::string GetS(stringindex_t index)
```

Comment:
Maybe return the const referrence?

Reply:
I agree that a const ref is usually a good idea. There are two reasons I wanted to use a string instead of a const ref:
- There are rare cases where refs could cause problems, especially in code that both gets and sets a string.
- If the string is unset, then GetS returns a newly constructed empty string. The alternative would be to extern declare a global empty string and return a const reference to it.

What do you think?

Reply:
I do mainly worry about the Autocode side which goes to use this widely (I want to make Autocode class hold string indexes and AutocodeManager operate with them too, and so, if Autocode wants to print any string, it will give the index, and it will be printed every frame, so, I want it should use the const reference to don't copy the string when it's not needed)

Reply:
I don't think that Autocode uses the level strings right now, but uses its own strings instead. But I have updated it to use const refs now.